staging/src/k8s.io/apiserver/plugin/pkg/audit/log/backend.go

```diff
@@ -17,77 +17,195 @@ limitations under the License.
 package log
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"strings"
+	"time"
 
+	"github.com/golang/glog"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
 	auditinternal "k8s.io/apiserver/pkg/apis/audit"
 	"k8s.io/apiserver/pkg/audit"
 )
 
 const (
+	// ModeDirect indicates that the backend should process events directly,
+	// Serialize event, then write it to logs. This mode spend more time.
+	ModeDirect = "direct"
```

Comment:
I think naming it `Blocking` and `NonBlocking` (or `Buffered`) will make it more clear for a user

Reply:
ok